### PR TITLE
 [QF-4567] feat: sync word-by-word content types to backend & improve settings drawer UI

### PR DESCRIPTION
**Pull request by yousefdergham**
# Summary

  - Sync wordByWordTooltipContentType and wordByWordInlineContentType to backend instead of only syncing wordByWordContentType
  - Auto-compute wordByWordDisplay from content types (no longer synced separately)
  - Add className prop to Section.Footer for custom styling
  - Add valueClassName prop to SelectionCard for higher specificity styling without :global() and !important
  - Remember last opened settings tab using Redux state
  - Add mobile padding for reciter drawer header

  Closes: https://quranfoundation.atlassian.net/browse/QF-4567

  ## Type of Change

  - 🐛 Bug fix (non-breaking change which fixes an issue)

   ##  Scope Confirmation

  - This PR addresses one feature/fix only
  - If multiple changes were needed, they are split into separate PRs

    ## Rollback Safety

  - Can be safely reverted without data issues or migrations
  - Rollback requires special steps (describe below):

    ## Test Plan

  - Unit tests added/updated
  - Integration tests added/updated
  - Manual testing performed

   ##  Testing steps:

  1. Login to the app
  2. Open Settings drawer → More tab → Word by Word section
  3. Toggle tooltip translation/transliteration checkboxes → verify syncs to backend with wordByWordTooltipContentType
  4. Toggle inline translation/transliteration checkboxes → verify syncs to backend with wordByWordInlineContentType
  5. Click Reset button → verify settings reset correctly via bulk endpoint
  6. Close and reopen settings drawer → verify last opened tab is remembered
  7. Test as guest user → verify local storage works without API calls
  8. Verify footer styling in Word by Word section (opacity, font size, border position)
  9. Open Reciter drawer on mobile → verify header padding is correct

  ##  Edge Cases Verified

  - ⏳ Loading state handled
  - ❌ Error state handled
  - 📭 Empty state handled
  - 👤 Logged-in vs guest behavior (if applicable)

   ##  Pre-Review Checklist

   ##  Code Quality

  - I have performed a self-review of my code (file by file)
  - My code follows the /.github/copilot-instructions.md
  - No any types used (or justified if unavoidable)
  - No unused code, imports, or dead code included
  - Complex logic has inline comments explaining "why"
  - Functions are under 30 lines and follow single responsibility

   ##  Testing & Validation

  - All tests pass locally (yarn test)
  - Build succeeds (yarn build)
  - Edge cases and error scenarios are handled

    ## Documentation

  - Code is self-documenting with clear naming
  - README updated (if adding features or setup changes)
  - Inline comments added for complex logic

    ## Localization (if UI changes)

  - All user-facing text uses next-translate
  - Only English locale files modified (Lokalise handles others)
  - RTL layout verified

    ## Accessibility (if UI changes)

  - Semantic HTML elements used
  - ARIA attributes added where needed
  - Keyboard navigation works

    ## Related PRs

  - Backend PR:  https://github.com/quran/quran.com-users-backend/pull/904

   ##  Reviewer Notes

  - Deployment order: Backend must be deployed first, then run backfill script, then deploy frontend
  - Frontend handles both old and new backend formats gracefully during transition
  - wordByWordContentType and wordByWordDisplay are deprecated but kept for backward compatibility
  - Removed :global() and !important from SelectionCard styling by adding valueClassName prop

  ##  AI Assistance Disclosure

  - AI tools were NOT used for this PR
  - AI tools were used, and I have thoroughly reviewed and validated all generated code

### CODE REVIEW

File: locales/en/quran-reader.json [hunk 1]
```diff
@@ -139,5 +139,7 @@
   "verse-preview-title": "Preview:",
   "verses-range": "Range of verses",
   "view-on-quran-reflect": "View on QuranReflect",
-  "wbw-helper-text": "Select options to show translation, transliteration, and/or recitation for each word."
+  "wbw-helper-text": "Select options to show translation, transliteration, and/or recitation for each word.",
+  "on-hover":"On Hover:",
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply locales/en/quran-reader.json]: JSON doesn't require spaces after colons - this is valid syntax and consistent with our other locale files.

File: src/components/Navbar/SettingsDrawer/SettingsBody.tsx [hunk 2]
```diff
@@ -44,15 +47,17 @@ const SettingsBody = () => {
...
+  // Update active tab when returning from a sub-view (e.g., Translation or Tafsir detail view)
   useEffect(() => {
-    const targetTab = getTabFromSettingsView(lastSettingsView);
-    setActiveTab(targetTab);
-  }, [lastSettingsView]);
+    if (lastSettingsView !== SettingsView.Body) {
+      const targetTab = getTabFromSettingsView(lastSettingsView);
+      handleTabChange(targetTab);
+    }
+  }, [lastSettingsView, handleTabChange]);
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/Navbar/SettingsDrawer/SettingsBody.tsx:68]: This is intentional behavior. When user navigates to Translation detail view and comes back, we want them to land on the Translation tab (not reset to Arabic). The useEffect ensures the tab stays contextually relevant based on which sub-view they were in.
